subworkflows/local/alignment_to_fastq.nf

```diff
@@ -0,0 +1,107 @@
+//
+// BAM/CRAM to FASTQ conversion, paired end only
+//
+
+include { SAMTOOLS_VIEW  as SAMTOOLS_VIEW_MAP_MAP      } from '../../modules/nf-core/samtools/view/main'
+include { SAMTOOLS_VIEW  as SAMTOOLS_VIEW_UNMAP_UNMAP  } from '../../modules/nf-core/samtools/view/main'
+include { SAMTOOLS_VIEW  as SAMTOOLS_VIEW_UNMAP_MAP    } from '../../modules/nf-core/samtools/view/main'
+include { SAMTOOLS_VIEW  as SAMTOOLS_VIEW_MAP_UNMAP    } from '../../modules/nf-core/samtools/view/main'
+include { SAMTOOLS_MERGE as SAMTOOLS_MERGE_UNMAP       } from '../../modules/nf-core/samtools/merge/main'
+include { SAMTOOLS_COLLATEFASTQ as COLLATE_FASTQ_UNMAP } from '../../modules/nf-core/samtools/collatefastq/main'
+include { SAMTOOLS_COLLATEFASTQ as COLLATE_FASTQ_MAP   } from '../../modules/nf-core/samtools/collatefastq/main'
+include { CAT_FASTQ                                    } from '../../modules/nf-core/cat/fastq/main'
+
+workflow ALIGNMENT_TO_FASTQ {
+    take:
+    input // channel: [meta, alignment (BAM or CRAM), index (optional)]
+    fasta // optional: reference file if CRAM format and reference not in header
+    fasta_fai
+
+    main:
+
+    ch_versions = Channel.empty()
+    // Index File if not PROVIDED -> this also requires updates to samtools view possibly URGH
+
+    // MAP - MAP
+    SAMTOOLS_VIEW_MAP_MAP(input, fasta, [])
```

Comment:
I'm pretty sure I'm just missing something obvious, but I don't get how the samtools flags are passed to the module. I guess they are part of the meta data somehow, but I can't seem to find where.

Reply:
They are specified